packages/tables/frontend/sirius-components-tables/src/rows/filters/useTableRowFilters.ts

```diff
@@ -0,0 +1,85 @@
+/*******************************************************************************
+ * Copyright (c) 2025 CEA LIST.
+ * This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v2.0
+ * which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *     Obeo - initial API and implementation
+ *******************************************************************************/
+
+import { gql, useQuery } from '@apollo/client';
+import { useMultiToast } from '@eclipse-sirius/sirius-components-core';
+import { useEffect, useState } from 'react';
+import { RowFilter } from './RowFiltersMenu.types';
+import {
+  GQLGetAllTableRowFiltersData,
+  GQLGetAllTableRowFiltersVariables,
+  GQLTableRowFilter,
+  UseTableRowFilterState,
+} from './useTableRowFilters.types';
+
+const getAllTableRowFiltersQuery = gql`
+  query getAllTableRowFilters($editingContextId: ID!, $representationId: ID!, $tableId: ID!) {
+    viewer {
+      editingContext(editingContextId: $editingContextId) {
+        representation(representationId: $representationId) {
+          description {
+            ... on TableDescription {
+              rowFilters(tableId: $tableId) {
+                id
+                label
+                defaultState
+              }
+            }
+          }
+        }
+      }
+    }
+  }
+`;
+
+export const useTableRowFilters = (editingContextId: string, tableId: string): UseTableRowFilterState => {
```

Comment:
It's odd for a hook to return it's internal state, the state is supposed to be internal not public. Especially when you have an unused XxxValue interface next to it :)